pkg/vsphere/simulator/folder.go

```diff
@@ -305,3 +305,54 @@ func (f *Folder) RegisterVMTask(c *types.RegisterVM_Task) soap.HasFault {
 
 	return r
 }
+
+func (f *Folder) MoveIntoFolderTask(c *types.MoveIntoFolder_Task) soap.HasFault {
+	task := CreateTask(f, "moveIntoFolder", func(t *Task) (types.AnyType, types.BaseMethodFault) {
+		for _, ref := range c.List {
+			obj := Map.Get(ref).(mo.Entity)
```

Comment:
Yes, we do this quite a bit in package simulator.  If the type is not what we expect, its a bug in the simulator or one's test code, in which case we want the panic.